commentlock.py

```diff
@@ -1,92 +1,124 @@
-    }
+    code = resp.status_code
 
-    print(csrf_token)
+    if code == 200:
+        return session
+    else:
+        raise ConnectionError(
+            f"Got status code {code}. Maybe you entered invalid login information?"
+        )
+
+def set_comments_allowed(
+    session: requests.Session,
+    username: str,
+    comments_allowed: bool,
+    cooldown: float = 0.1,
+):
 
-    # Get session data
-    session = s.get("https://scratch.mit.edu/session/", headers=headers)
-    print(session.status_code)
-    print(session.text)
-    
     offset = 0
-    
-    projects = {}
-    
-    while(len(projects) == 20 or offset == 0):
-        print("Page:" + str(math.floor(offset/20) + 1))
-    
-        projects_api = s.get("https://api.scratch.mit.edu/users/" + username + "/projects/?offset=" + str(offset), headers=headers)
-        
-        projects = json.loads(projects_api.text)
-        
-        for i in projects:
-            r = s.put("https://api.scratch.mit.edu/projects/" + str(i["id"]) + "/", headers=headers, json={"comments_allowed": comments_on})
-            print("https://api.scratch.mit.edu/projects/" + str(i["id"]))
-            print(r.status_code)
-    
-        offset += 20
\ No newline at end of file
+    projects = []
+
+    # Set up payload only once
+    payload = {"comments_allowed": comments_allowed}
+
+    while len(projects) == 20 or offset == 0:
+
+        # Use python integer division and f-strings to condense
+        print(f"Page: {offset // 20 + 1}")
+
+        # Use response.json() to get data immediately
+        projects = session.get(
+            f"https://api.scratch.mit.edu/users/{username}/projects/?offset={offset}"
+        ).json()
+
+        for project in projects:
+            # More f-strings :)
+            resp = session.put(
+                f"https://api.scratch.mit.edu/projects/{project['id']}/", json=payload
+            )
+
+            # Print out project title as well as id with some nice formatting.
+            success_text = (
+                "Success!"
+                if resp.status_code == 200
+                else f"Failed with code {resp.status_code}"
+            )
+            print(f"   {project['title']} [{project['id']}] - {success_text}")
+
+            # Use sleep to prevent ratelimiting / bans :D
+            sleep(cooldown)
+
+        offset += 20
+
+
+# Use __name__ == '__main__' to prevent accidental running
+if __name__ == "__main__":
+
+    print("Welcome to the commentlock tool!")
+
+    username = input("Enter your username: ")
+    password = getpass("Enter your password: ")
+
+    # Make comments_on variable into 1-line
+    comments_on = input("Should comments be enabled? Y/n: ").lower() == "y"
+
+    # Add cooldown
+    cooldown = float(
+        input("How long should the request cooldown be? (0.1 recommended): ")
+    )
+
+    print("\n")

```

Comment:
Is there a reason that this is on a different line?

Reply:
Nope! If you want to combine the lines, feel free. I used the code stylizer, `black` to do all of the stylizing after I wrote the code, so it wasn't intentional.